tests/test_external_roundtrip.rs

```diff
@@ -0,0 +1,109 @@
+use std::io::Read;
+use std::path::PathBuf;
+use std::process::Command;
+
+extern crate env_logger;
+#[macro_use]
+extern crate log;
+
+const DEBUG: bool = true;
+
+fn get_path(components: &[&str]) -> PathBuf {
+    let mut root = std::env::temp_dir();
+    for component in components {
+        root.push(component)
+    }
+    root
+}
+
+fn run(mut command: std::process::Command) -> Result<(), std::io::Error> {
+    let output = command.output()?;
+    if DEBUG || !output.status.success() {
+        debug!(target: "test", "Command: {command:?}\n** out:{out}\n** err:{err}",
+            command = command,
+            out =  String::from_utf8(output.stdout).unwrap_or_else(|_| "<cannot decode output>".to_string()),
+            err =  String::from_utf8(output.stderr).unwrap_or_else(|_| "<cannot decode output>".to_string())
+        );
+    }
+    assert!(output.status.success());
+    Ok(())
+}
+
+fn perform_test(prefix: &str, source: &str) {
+    let path_in_source = std::path::Path::new(source).to_path_buf();
+    let path_tmp_source = get_path(&[prefix, "sample.js"]);
```

Comment:
I think I'd rather use [`tempdir`](https://docs.rs/tempdir) or [`tempfile`](https://docs.rs/tempfile) here which would take care both of creating a directory name that is guaranteed to be unique / not already existing on the filesystem, and also would clean up after themselves rather than leaving a stray directory in the temp folder.

Reply:
Alternatively, with `tempfile` you can just create unique temporary files with the same effect, but not having to do any Path manipulations.